lib/metaGitUpdate.js

```diff
@@ -8,11 +8,20 @@ const util = require('util');
 module.exports = (options = {}) => {
   const meta = getMetaFile({ confirmInMetaRepo: true, warn: false });
   if (!meta) return;
+  
+  // Filter out existing projects but allow empty directories
+  const missing = Object.keys(meta.projects).filter(name => {
+    if (fs.existsSync(name)) {
+      const stat = fs.lstatSync(name);
+      if (stat.isDirectory()) {
+        const files = fs.readdirSync(name);
+        return files.length === 0;
+      }
+      return false;
+    }
+    return true;
+  });
```

Comment:
Would it be better to allow any empty directories, only if they happen to be listed in the paths specified in .gitmodules? You could use something like the following to get the list of submodules and then compare those within your .filter above. 

```
const fs = require('fs').promises;

async function getSubmodulePaths(gitmodulesPath = '.gitmodules') {
  try {
    // Read the .gitmodules file
    const content = await fs.readFile(gitmodulesPath, 'utf8');

    // Parse the content using regular expressions
    const submoduleRegex = /\[submodule "([^"]+)"\]\s*path = (.+)/g;
    const submodulePaths = [];
    let match;

    while ((match = submoduleRegex.exec(content)) !== null) {
      const path = match[2].trim();
      submodulePaths.push(path);
    }

    return submodulePaths;
  } catch (error) {
    console.error('Error reading or parsing .gitmodules file:', error);
    return [];
  }
}

// Usage
getSubmodulePaths()
  .then(paths => console.log('Submodule paths:', paths))
  .catch(error => console.error('Error:', error));
```

Reply:
Great add.  Updated with some modifications.  Defaults to empty to not break existing repos without `.gitmodules`.  Also made the function fully sync, so it doesn't pollute the plugin with async up the thread.